Set v6 default payload compression to w19T0.zstdio

ffesti opened the pull request:
Keep the compression for source packages untouched to keep maximum forward compatibility for those.

This also enables multi-threaded compression per default.

Resolves: #3312

pmatilai commented:
Don't enable threads, we're nowhere near ready to handle that.

pmatilai commented:
This makes me wonder why we're not supplying an actual default in the macros file. We have the source/binary differentation there, so we could just change the default macro and not necessarily have to touch the code logic at all.

ffesti commented:
Well, we don't have the means to differentiate between v4 and v6 there.

ffesti commented:
I think we should add a default that distros can just stick with and drop their own changes to the macro file.

Having the expression in the macro files would have the benefit of teaching the users how to do different values for v4 and v6. So that's probably something worth adding.

ffesti commented:
Switched to a expression in macros.in. Also dropped the multi-threaded compression. We really need to get our stuff together some day so we can enable that.

dralley commented on macros.in:
```diff
@@ -380,10 +380,11 @@ Supplements:   (%{name} = %{version}-%{release} and langpacks-%{1})\
 #
 #%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_binary_payload	%[ %_rpmformat >=6 ? "w19.zstdio" : "w9.gzdio" ]
```
Why make it conditional?  All versions of Fedora (+ RHEL 9 and derivatives) have been building (binary) packages with zstd payloads for years at this point.  As far as I can tell just about every existing supported distribution has a version of RPM that supports zstd and builds RPM with zstd support enabled.  EL8+, SLES 15, etc.

The list of distros that would choke on a zstd package is... EL7 Extended Lifecycle, and maybe SLES 12 LTSS?  

I would think it would be OK to use zstd even for source packages at this point.

pmatilai replied:
We try to represent a relatively independent upstream. For v4 packages, the default payload compression has been gzip for the ~20 years of its existence, we don't want to change that *now*.

pmatilai replied:
You'd be surprised how much people expect to do just that sort of thing. 
But it's not so much about compatibility, and more about just not being worth it: source tarballs are practically always compressed already and burning extra cpu on a better compression algorithm isn't going to make the damnest of differences. That's why nobody updated it for v4 either, whereas for binaries nobody used gzip for ages.

pmatilai commented:
> Also dropped the multi-threaded compression. We really need to get our stuff together some day so we can enable that.

Be my guest :smiling_imp: It's fun times coordinating the resources between multiple different threading systems, each unaware of the others.

pmatilai commented:
This of course still doesn't solve the issue of existing distro defaults overriding v6 settings too, but probably not important to get hung over just now. At least the macro version is very, very very simple.
